pkg/mempool/simplemempool/internal/parts/computeids/computeids.go

```diff
@@ -19,13 +19,27 @@ import (
 func IncludeComputationOfTransactionAndBatchIDs(
 	m dsl.Module,
 	mc common.ModuleConfig,
-	_ *common.ModuleParams,
+	params *common.ModuleParams,
 	_ *common.State,
 ) {
 	mppbdsl.UponRequestTransactionIDs(m, func(txs []*trantorpbtypes.Transaction, origin *mppbtypes.RequestTransactionIDsOrigin) error {
-		txMsgs := make([]*hasherpbtypes.HashData, len(txs))
-		for i, tx := range txs {
-			txMsgs[i] = &hasherpbtypes.HashData{Data: serializeTXForHash(tx.Pb())}
+		if txs == nil || len(txs) > params.MaxTransactionsInBatch {
+			// Invalid request, ignore
+			return nil
+		}
+
+		txMsgs := make([]*hasherpbtypes.HashData, 0, len(txs))
+		for _, tx := range txs {
+			serializedTx := serializeTXForHash(tx.Pb())
+			if serializedTx == nil {
```

Comment:
Also not sure if this can occur without an error in our code... (Not claiming it can't but I'm not sure.) In such a case, silently ignoring the transaction is very dangerous. Not to mention that it seems like a list of non-nil transactions would always have the same hash, regardless of how it is interleaved with nil transactions.
I think it would be better to return errors all through the mempool implementation whenever there is a nil transaction. We'd then only need to sanitize the input.

Reply:
I think what you mean is then just halting this calculation and returning nil (because this can be triggered by an external event [UponProvideBatchMessageReceived](https://github.com/filecoin-project/mir/blob/8f352300a01bb71d1bc05cb62a92e0e105ad0862/pkg/availability/multisigcollector/internal/parts/batchreconstruction/batchreconstruction.go#L128-L142).

Reply:
Done what I mentioned but feel free to comment.

Reply:
What I meant was to
1. Return an error here
2. Make sure that this event is never triggered with `nil`s in the transaction list. (Might be less efficient, but is cleaner.) Should that be about to happen because of an external message, catch it right there and ignore the message with a warning.

Reply:
This relates to the previous comment. Sure I can make those checks in [UponProvideBatchMessageReceived](https://github.com/filecoin-project/mir/blob/8f352300a01bb71d1bc05cb62a92e0e105ad0862/pkg/availability/multisigcollector/internal/parts/batchreconstruction/batchreconstruction.go#L128-L142), but the one for MaxTransactionsInBatch I am reluctant to move for as long as that parameter is part of the default config of the mempool only.